src/accelerated_image_processor_common/include/accelerated_image_processor_common/processor.hpp

```diff
@@ -0,0 +1,107 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "accelerated_image_processor_common/datatype.hpp"
+#include "accelerated_image_processor_common/parameter.hpp"
+
+#include <functional>
+#include <optional>
+#include <string>
+#include <utility>
+
+namespace accelerated_image_processor::common
+{
+/**
+ * @brief Base class for image processors.
+ */
+class BaseProcessor
+{
+public:
+  // @brief Post-processing function type.
+  using PostProcessFn = std::function<void(common::Image &)>;
+
+  /**
+   * @brief Constructor.
+   */
+  explicit BaseProcessor(ParameterMap parameters) : parameters_(std::move(parameters)) {}
+
+  virtual ~BaseProcessor() = default;
+
+  /**
+   * @brief Set the post-processing function and return a reference to the current object.
+   * @param postprocess_fn The post-processing function.
+   */
+  void with_postprocess(const PostProcessFn & postprocess_fn) { postprocess_fn_ = postprocess_fn; }
+
+  /**
+   * @brief Process the input image.
+   * @param image The input image.
+   */
+  void process(common::Image & image)
+  {
+    auto processed = this->process_impl(image);
+    if (postprocess_fn_) {
+      postprocess_fn_.value()(processed);
```

Comment:
```suggestion
      postprocess_fn_(processed);
```
Since `postprocess_fn_` is raw std::function, I believe `.value()` is not neccesary